src/jit/lowerxarch.cpp

```diff
@@ -307,6 +307,7 @@ void Lowering::TreeNodeInfoInit(GenTree* stmt)
                         GenTreeLclVarCommon* lclVarPtr = tree->gtOp.gtOp1->AsLclVarCommon();
                         LclVarDsc* varDsc = &(compiler->lvaTable[lclVarPtr->gtLclNum]);
                         assert(varDsc->lvIsMultiRegArgOrRet);
+                        varDsc->lvDoNotEnregister = true;
```

Comment:
No, I don't think so. I think that we would need to settle on a DNER reason for all such cases, and it should be less fully general than DNER_IsStruct. I think this is a reasonable near-term fix.